api/receivers.py

```diff
@@ -0,0 +1,236 @@
+import json
+from django.db import transaction
+from django.db.models.signals import pre_delete
+from django.dispatch import receiver
+from reversion.models import Revision, Version
+from reversion.signals import post_revision_commit
+from api.models import ReversionDifferenceLog, User, Country, Event
+from deployments.models import DeployedPerson
+from per.models import Form
+from middlewares.middlewares import get_username
+
+
+MODEL_TYPES = {
+    'api.action': 'Action',
+    'api.appeal': 'Appeal',
+    'api.appealdocument': 'Appeal Document',
+    'api.country': 'Country',
+    'api.cronjob': 'CronJob',
+    'api.district': 'District',
+    'api.emergencyoperationsdataset': 'Emergency Operations Dataset',
+    'api.emergencyoperationsea': 'Emergency Operations Emergency Appeals',
+    'api.emergencyoperationsfr': 'Emergency Operations Final Reports',
+    'api.emergencyoperationspeoplereached': 'Emergency Operations People Reached',
+    'api.event': 'Emergency',
+    'api.fieldreport': 'Field Report',
+    'api.gdacsevent': 'GDACS Event',
+    'api.profile': 'User Profile',
+    'api.region': 'Region',
+    'api.situationreport': 'Situation Report',
+    'api.situationreporttype': 'Situation Report Type',
+    'deployments.eruowner': 'ERU Owner',
+    'deployments.erureadiness': 'ERU Readiness',
+    'deployments.partnersocietyactivities': 'Partner society activity',
+    'deployments.partnersocietydeployment': 'Partner society deployment',
+    'deployments.personnel': 'Personnel',
+    'deployments.personneldeployment': 'Personnel Deployment',
+    'deployments.project': 'Project',
+    'deployments.regionalproject': 'Regional project',
+    'notifications.subscription': 'Subscription',
+    'notifications.surgealert': 'Surge alert',
+    'per.form': 'PER Form',
+    'per.formdata': 'PER Form Data',
+    'per.nicedocument': 'PER Document',
+    'per.nsphase': 'PER NS Phase',
+    'per.overview': 'PER General Overview',
+    'per.workplan': 'PER Work Plan',
+    'registrations.pending': 'Pending registration',
+}
+
+
+def get_object_name(otype, obj):
```

Comment:
Can you add some comments to explain what this function does and why, please? Then we can discuss if there's probably a way to do this more declaratively that avoids these long if statements with hard-coded values that can be hard to read / modify.

Reply:
So this only gets the objects' `name` - which differs from model to model - for the global log page (which I called 'Reversion difference logs'). Since there is no switch-case statement in Python afaik, I had to go with the ugly and long if-elif-else solution.

Straight from reversion the model names come like `api.profile` but since reversion doesn't really cover the logging of deletions I had to create that manually and from that the model name comes like `profile` thus why I had to handle both. I agree that this is not the most maintainable version but I couldn't figure out how could I get a "globalized" model name and/or an object name for each different log in a prettier (standard) way.

I'd glad to fix this if you have any suggestions please don't hold it back 😄